src/components/[guild]/activity/ActivityLogContext.tsx

```diff
@@ -61,24 +68,38 @@ const transformActivityLogInfiniteResponse = (
   return transformedResponse
 }
 
-const ActivityLogContext = createContext<
-  Omit<SWRInfiniteResponse<ActivityLogActionResponse>, "mutate" | "data"> & {
-    data: ActivityLogActionResponse
-    mutate: () => void
-    isUserActivityLog: boolean
-  }
->(undefined)
+type ActivityLogContextType = Omit<
+  SWRInfiniteResponse<ActivityLogActionResponse>,
+  "mutate" | "data"
+> & {
+  data: ActivityLogActionResponse
+  mutate: () => void
+  isUserActivityLog: boolean
+  actionGroup: ActivityLogActionGroup
+  setActionGroup: (value: ActivityLogActionGroup) => void
+  withActionGroups: boolean
+}
+
+const ActivityLogContext = createContext<ActivityLogContextType>(undefined)
 
 type Props = {
   withSearchParams?: boolean
   isInfinite?: boolean
   limit?: number
+  withActionGroups?: boolean
 } & OneOf<{ userId: number }, { guildId: number }>
 
+const getAdminActions = () => {
```

Comment:
Let's just move the return value of this function in `constants.ts`, so we don't need to filter this object every time)

```ts
const ADMIN_ACTIONS = Object.values(ACTION).filter(
    (action) => !USER_ACTIONS.includes(action) && !HIDDEN_ACTIONS.includes(action)
)
```